scripts/get_ngram_num.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+
+# we're using python 3.x style print but want it to work in python 2.x,
+from __future__ import print_function
+import re, os, argparse, sys, math, warnings, subprocess, shutil, tempfile, threading
+from collections import defaultdict
+
+parser = argparse.ArgumentParser(description="This script get the total number of n-grams "
+                                 "in a pocolm 'lm-dir' (as validated by validate_lm_dir.py).")
+
+parser.add_argument("lm_dir_in",
+                    help="Source directory, for the input language model.")
+
+
+args = parser.parse_args()
+
+# Add the script dir and the src dir to the path.
+os.environ['PATH'] = (os.environ['PATH'] + os.pathsep +
+                      os.path.abspath(os.path.dirname(sys.argv[0])) + os.pathsep +
+                      os.path.abspath(os.path.dirname(sys.argv[0])) + "/../src");
+# this will affect the program "sort" that we call.
+os.environ['LC_ALL'] = 'C'
+
+
+if os.system("validate_lm_dir.py " + args.lm_dir_in) != 0:
+    sys.exit("get_ngram_num.py: failed to validate input LM-dir")
+
+
+num_splits = None
+
+if os.path.exists(args.lm_dir_in + "/num_splits"):
+    f = open(args.lm_dir_in + "/num_splits")
+    num_splits = int(f.readline())
+    f.close()
+
+tot_num_gram = 0
+
+def GetNumNgram(split_index):
+    if split_index == None:
```

Comment:
It shouldn't be necessary to use this script (which is very inefficient).
In prune_lm_dir.py, see the following lines:
```
        [ tot_ngrams, shadowed, protected, pruned ] = p.stdout.readline().split()
...
        final_num_ngrams = int(tot_ngrams) - int(pruned)
```
tot_ngrams is the number of n-grams before pruning, and final_num_ngrams is after.  
I believe this will give you everything you need.